ircd/reject.c

```diff
@@ -250,6 +250,13 @@ check_reject(rb_fde_t *F, struct sockaddr *addr)
 			rb_setselect(F, RB_SELECT_WRITE | RB_SELECT_READ, NULL, NULL);
 			if(rdata->aconf)
 			{
+				if(rdata->aconf->status & CONF_ILLEGAL)
+				{
+					rb_dlinkDelete(&rdata->rnode, &reject_list);
+					reject_free(rdata);
+					rb_patricia_remove(reject_tree, pnode);
+					return 0;
```

Comment:
Doesn't this also need to decrement `ServerStats.is_rej`? (Since it was unconditionally incremented above)

Reply:
Good catch, but I'd rather just move that below the ifs